src/core/helpers/formatting.ts

```diff
@@ -0,0 +1,75 @@
+import { MessageEntity, User } from 'typegram'
+
+export interface FmtString {
+  text: string
+  entities?: MessageEntity[]
+  parse_mode?: undefined
+}
+
+export class FmtString implements FmtString {
+  constructor(public text: string, public entities?: MessageEntity[]) {
+    // force parse_mode to undefined if entities are present
+    if (entities) this.parse_mode = undefined
+  }
+  static normalise(content: string | FmtString) {
+    if (typeof content === 'string') return new FmtString(content)
+    return content
+  }
+  toString() {
+    return this.text
+  }
```

Comment:
Unnecessary?
```suggestion
```

Reply:
https://github.com/telegraf/telegraf/blob/18a88154efb3b3fdd84cc988c2382c7724d771b7/src/core/helpers/formatting.ts#L71

Used here to convert `string | FmtString` to `string` without type-checking.

Reply:
You already `normalise` in the next line :unamused:

Reply:
Lol, you're right. Fixed in f70829c729a348b2130c1284c3fd35e66a7bb3e5